decompiler/structures/pseudo/delogic_logic.py

```diff
@@ -0,0 +1,159 @@
+"""Implements translating pseudo instructions into delogic statements."""
+from __future__ import annotations
+
+from typing import Generic, Union
+
+from simplifier.world.nodes import Operation as WorldOperation
+from simplifier.world.nodes import Variable as WorldVariable
+from simplifier.world.nodes import WorldObject
+from simplifier.world.world import World
+
+from .expressions import Constant, Expression, Variable
+from .instructions import Branch
+from .logic import BaseConverter
+from .operations import Condition, Operation, OperationType
+
+
+class DelogicConverter(BaseConverter):
+    """Class in charge of converting psudo expressions into Logic statements"""
+
+    def __init__(self):
+        self._world = World()
+        self._var_count = 0
+
+    @staticmethod
+    def get_world():
+        """Retrieve a World object."""
+        return World()
+
+    def convert(self, expression: Union[Expression, Branch], w: World = None) -> WorldObject:
+        """Convert a given expression or branch into a logic statement of type WorldObject.
+        w can be obtained from LogicConverter.get_world()"""
+        if isinstance(expression, Variable):
+            return self._convert_variable(expression, self._world if w is None else w)
+        if isinstance(expression, Constant) and (isinstance(expression.value, int) or isinstance(expression.value, float)):
+            return self._convert_constant(expression, self._world if w is None else w)
+        if isinstance(expression, Branch):
+            return self._convert_branch(expression, self._world if w is None else w)
+        if isinstance(expression, Condition):
+            return self._convert_condition(expression, self._world if w is None else w)
+        if isinstance(expression, Operation):
+            return self._convert_operation(expression, self._world if w is None else w)
+        raise ValueError(f"Could not convert {expression} into a logic statement.")
+
+    def make_variable(self, expr: WorldObject) -> WorldObject:
```

Comment:
A better name would perhaps be `define_expression` or `define_formula`. And the return value is always a Variable.

Reply:
I think I'll go with ```define_expression```!